[wasm][testing] hosting echo server in xharness process

pavelsavara opened the pull request:
This makes WebSocket test on WASM easier to work with, by moving the tests to inner loop. In order to do that we are hosting the echo test server which the tests use, in the xharness process. Because wasm is always tested with xharness.

- build echo test server and pass it to xharness
- move networking tests to inner loop for browser
    - 110 tests enabled in System.Net.Http.Functional.Tests
    - 72 tests enabled in System.Net.WebSockets.Client.Tests
- discovers https://github.com/dotnet/runtime/issues/53591
- discovers https://github.com/dotnet/runtime/issues/53592
- discovers https://github.com/dotnet/runtime/issues/53872
- discovers https://github.com/dotnet/runtime/issues/53874
- discovers https://github.com/dotnet/runtime/issues/53876
- fixes https://github.com/dotnet/runtime/issues/53018
- fixes https://github.com/dotnet/runtime/issues/53668

mdh1418 commented on src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj:
```diff
@@ -1,7 +1,7 @@
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>$(AspNetCoreAppCurrent)</TargetFramework>
```
Why `<TargetFramework>$(AspNetCoreAppCurrent)</TargetFramework>`  over `<TargetFramework>$(NetCoreAppCurrent)</TargetFramework>` or `<TargetFramework>$(NetCoreAppCurrent)</TargetFramework>`? 
Though they evaluate to the same value at the moment, it seems like `AspNetCoreAppCurrent` is only used by the wasm debugger, so it "feels" like one of the other properties should be used here but I'm not sure

pavelsavara commented:
`fail: Failed to find the middleware assembly at /__w/1/s/artifacts/bin/NetCoreServer/net6.0-Release/NetCoreServer.dll`

mdh1418 commented on docs/workflow/testing/libraries/testing-wasm.md:
```diff
@@ -100,12 +100,15 @@ The following shows how to run tests for a specific library
 
     Example: `WasmXHarnessMonoArgs="--runtime-arg=--trace=E"`
 
+- `$(WasmXHarnessMonoArgs)` - env variables to mono
```
Should this be folded into L105 because it is still the same variable? Perhaps `arguments/env variables to mono` with both examples.

pavelsavara replied:
fixed

mdh1418 commented on eng/testing/tests.props:
```diff
@@ -32,6 +32,23 @@
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
+    <ProjectReference Include="$(CommonTestPath)System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj" 
+      Condition="'$(TargetOS)' == 'Browser'" ReferenceOutputAssembly="false"/>
```
Redundant Condition?

pavelsavara replied:
fixed

mdh1418 commented on eng/testing/tests.props:
```diff
@@ -32,6 +32,23 @@
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
+    <ProjectReference Include="$(CommonTestPath)System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj" 
```
Is this primarily here because of L40 that may point `_TestEchoMiddleware` at `NetCoreServer`? So this isn't used when its a `ContinuousIntegrationBuild`?

pavelsavara replied:
This is here to make sure it's built. But none of the test use that dependency inside WASM process. It's needed both for CI and locally.

mdh1418 commented on src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs:
```diff
@@ -344,10 +344,10 @@ public async Task SendAsync_RequestHeaderInResponse_Success(string name, string
         }
 
         [OuterLoop("Uses external servers")]
+        [SkipOnPlatform(TestPlatforms.Browser, "Host is ignored on Browser")]
```
What does this mean? Is `m.Headers.Host` problematic? Will it always be ignored on Browser or is this something that can be implemented to not be ignored?

pavelsavara replied:
This is owned by underlying http implementation in browser. It could not be changed AFAIK.

radical replied:
I'm not clear on why we are replacing the `ActiveIssue` with this.

mdh1418 commented on src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs:
```diff
@@ -148,14 +148,14 @@ public async Task GetAsync_SetCookieContainerMultipleCookies_CookiesSent()
             foreach (Cookie c in cookies)
             {
-                cookieContainer.Add(Configuration.Http.Http2RemoteEchoServer, c);
+                cookieContainer.Add(Configuration.Http.Http2SecureRemoteEchoServer, c);
```
Why the change from `Http2RemoteEchoServer` to `Http2SecureRemoteEchoServer`?

mdh1418 replied:
For this and other related changes

pavelsavara replied:
Because it's actually used as HTTPS, currently wrongly named. In our case it also matters to use HTTPS protocol to reach correct port number, which is set by the environment variable.

mdh1418 commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -789,14 +826,23 @@ public async Task PostAsync_ReuseRequestContent_Success(Configuration.Http.Remot
             string method,
             Uri serverUri)
         {
+            if (method == "TRACE" && PlatformDetection.IsBrowser)
```
Why modify the ActiveIssues to be of this form? If the problem occurs when method is `TRACE` I think the issue can reflect that instead. The ActiveIssue attribute should skip this method from being ran, and that would reflect in the test results. I'm not sure, but would the early return indicate this test method as passing (which sounds like it shouldn't be)?

mdh1418 replied:
Same for the issue where method is `HEAD`

pavelsavara replied:
Because this is data driven test, there are many combinations. Multiple endpoints/protocols x multiple HTTP verbs.
Most of them work well and we wouldn't like to disable them too. 

Yes, this reports false positive. This is how we deal with complex conditions elsewhere too. https://github.com/dotnet/runtime/blob/1e1951825bb58417b0566e4b641544483c9e8865/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs#L186

mdh1418 commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -759,13 +791,18 @@ public async Task PostAsync_ReuseRequestContent_Success(Configuration.Http.Remot
             string method,
             Uri serverUri)
         {
+            if (method == "GET" && PlatformDetection.IsBrowser)
```
Same question as https://github.com/dotnet/runtime/pull/52923/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csproj&file-filters%5B%5D=.md&file-filters%5B%5D=.proj#r645655655

pavelsavara replied:
Same answer

mdh1418 commented on src/libraries/Common/tests/System/Net/Configuration.Http.cs:
```diff
@@ -53,32 +54,34 @@ public static partial class Http
             public static readonly Uri RemoteVerifyUploadServer = new Uri("http://" + Host + "/" + VerifyUploadHandler);
             public static readonly Uri SecureRemoteVerifyUploadServer = new Uri("https://" + SecureHost + "/" + VerifyUploadHandler);
-            public static readonly Uri Http2RemoteVerifyUploadServer = new Uri("https://" + Http2Host + "/" + VerifyUploadHandler);
+            public static readonly Uri Http2RemoteVerifyUploadServer = new Uri("https://" + Http2SecureHost + "/" + VerifyUploadHandler);
```
Following the naming convention, shouldn't the name be `Http2SecureRemoteVerifyUploadServer` so its clear in that wherever the variable is used that it references `Http2SecureHost`
Same for L69 (`Http2SecureRemoteDeflateServer`), L70 (`Http2SecureRemoteGZipServer`), L76 `Http2SecureServers`).

I take it that all the tests that use these variables were intended to use the secure form, and so we no longer need the non-secure forms right?

pavelsavara replied:
Your logic is correct, but it would propagate to too many places. I would keep it as is.

radical commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -95,9 +94,8 @@ public async Task SendAsync_SimpleGet_Success(Configuration.Http.RemoteServer re
-        [OuterLoop("Uses external servers")]
+        [OuterLoop("Uses external servers", TestPlatforms.Any & ~TestPlatforms.Browser)]
         [Theory, MemberData(nameof(RemoteServersMemberData))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/53018", TestPlatforms.Browser)]
         public async Task SendAsync_MultipleRequestsReusingSameClient_Success(Configuration.Http.RemoteServer remoteServer)
```
Do I understand this correctly? :
- Runs the tests on OuterLoop for non-browser cases
- Runs them with the regular builds for browser

radical commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -344,16 +335,25 @@ public async Task SendAsync_HttpRequestMsgResponseHeadersRead_StatusCodeOK(Confi
                 var content = new StringContent(data, Encoding.UTF8);
-                content.Headers.ContentMD5 = TestHelper.ComputeMD5Hash(data);
+
+                if(PlatformDetection.IsBrowser)
+                {
+                    // [ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]
+                    content.Headers.Add("Content-MD5-Skip", "browser");
+                }
+                else
```
Why is this preferable over not running the test for browser at all?

pavelsavara replied:
We don't have MD5, but there are other features the test is testing, not just payload integrity.

mdh1418 reviewed (approved):
Overall looks good to me, thanks!

The remaining nits I have are the naming in `Configuration.Http.cs` and using `NetCoreAppCurrent` instead in `src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj`

radical commented on src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs:
```diff
@@ -1226,7 +1226,7 @@ public async Task SendAsync_CorrectVersionSelected_ExternalServer(Version reques
             if (serverVersion == HttpVersion.Version20)
             {
-                remoteServer = useSsl ? RemoteHttp2Server : null;
+                remoteServer = useSsl ? RemoteSecureHttp2Server : RemoteHttp2Server;
```
Doesn't this change the test? The test wasn't doing anything in case of `!useSsl`, right?

pavelsavara replied:
Now it's running the same test over plan HTTP also. Originally `RemoteHttp2Server` had `https://` protocol in it and bad name. Do you prefer to not run it ?

radical replied:
> Do you prefer to not run it ?

I'm just wondering why we are changing the test to do something it wasn't doing earlier, which is run for `http://`. If you want to run it over that too, then that should be a different PR.

pavelsavara replied:
ok, reverted the change

radical commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -836,9 +882,8 @@ public static IEnumerable<object[]> SendAsync_SendSameRequestMultipleTimesDirect
         }
 
-        [OuterLoop("Uses external servers")]
+        [OuterLoop("Uses external servers", TestPlatforms.Any & ~TestPlatforms.Browser)]
```
I would suggest adding a `PlatformDetection.IsLocalEchoServerAvailable`, which can be true for browser. Then we can condition `OuterLoop` on `!PlatformDetection.IsLocalEchoServerAvailable`

pavelsavara replied:
Good idea to refactor this, will do, thanks. It's not bool but `TestPlatforms` enum.

radical commented on src/libraries/Common/tests/System/Net/Http/PostScenarioTest.cs:
```diff
@@ -262,7 +249,15 @@ public async Task PostAsync_EmptyContent_ContentTypeHeaderNotSent(Configuration.
-                    requestContent.Headers.ContentMD5 = TestHelper.ComputeMD5Hash(requestBody);
+                    if(PlatformDetection.IsBrowser)
+                    {
+                        // [ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]
+                        requestContent.Headers.Add("Content-MD5-Skip", "browser");
```
Why modify the tests for this issue, instead of skipping for now?

pavelsavara replied:
It was blocking most POST tests. They are more useful in this reduced form, than not running it at all.

radical replied:
IMHO, these test changes should be in a separate PR. This one is enabling tests that depend on echo servers, IIUC.

pavelsavara commented on eng/testing/tests.props:
```diff
@@ -32,6 +32,23 @@
   </PropertyGroup>
 
+  <!-- when testing with Xharness we could host echo test server in the xharness process and avoid external server dependency -->
+  <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
```
moved

pavelsavara commented on eng/testing/tests.props:
```diff
@@ -32,6 +32,23 @@
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
+    <ProjectReference Include="$(CommonTestPath)System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj" 
```
Moved

pavelsavara commented on src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:
```diff
@@ -77,8 +77,7 @@ public async Task UseDefaultCredentials_SetToFalseAndServerNeedsAuth_StatusCodeU
 
-        [OuterLoop("Uses external servers")]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]
+        [OuterLoop("Uses external servers", TestPlatforms.Any & ~TestPlatforms.Browser)]
```
For the browser test I pass `Content-MD5-Skip` and because I could handle it differently in xharness hosted server, this makes lot of tests pass. All those tests are testing other useful things besides integrity of the payload.

pavelsavara commented:
https://github.com/dotnet/runtime/pull/52923#discussion_r645689216
`[SkipOnPlatform(TestPlatforms.Browser, "Host is ignored on Browser")]` do you believe we could fix this @radical ? Then `ActiveIssue` would be more appropriate.

radical commented:
> [#52923 (comment)](https://github.com/dotnet/runtime/pull/52923#discussion_r645689216)
> `[SkipOnPlatform(TestPlatforms.Browser, "Host is ignored on Browser")]` do you believe we could fix this @radical ? Then `ActiveIssue` would be more appropriate.

As @mdh1418 asked, `What does this mean? Is m.Headers.Host problematic?`, could you just explain the reasoning? What fails, and how does it fail?

radical commented on src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs:
```diff
@@ -35,31 +35,40 @@ public static async Task InvokeAsync(HttpContext context)
-                if (expectedHash == actualHash)
+                string skip = context.Request.Headers["Content-MD5-Skip"];
+                if(!string.IsNullOrEmpty(skip)) 
                 {
                     context.Response.StatusCode = 200;
                 }
                 else
                 {
-                    context.Response.StatusCode = 400;
-                    context.Response.SetStatusDescription("Received request body fails MD5 checksum");
+                    // Get expected MD5 hash of request body.
+                    string expectedHash = context.Request.Headers["Content-MD5"];
+
+                    if (string.IsNullOrEmpty(expectedHash))
+                    {
+                        context.Response.StatusCode = 400;
+                        context.Response.SetStatusDescription("Missing 'Content-MD5' request header");
+                        return;
+                    }
+
+                    // Compute MD5 hash of received request body.
+                    string actualHash;
+                    using (MD5 md5 = MD5.Create())
+                    {
+                        byte[] hash = md5.ComputeHash(requestBodyBytes);
+                        actualHash = Convert.ToBase64String(hash);
+                    }
+
+                    if (expectedHash == actualHash)
+                    {
+                        context.Response.StatusCode = 200;
+                    }
+                    else
+                    {
+                        context.Response.StatusCode = 400;
+                        context.Response.SetStatusDescription("Received request body fails MD5 checksum");
+                    }
```
Could we restrict this PR to just enabling tests that depend on the echo server, as the PR title says? If they need more changes, then that should go into a separate PR, so that it can get reviewed properly, IMHO.

pavelsavara committed:
- improve doc
- move tests to inner loop
- more granular ActiveIssue https://github.com/dotnet/runtime/issues/53592 for lack of TRACE
- more granular ActiveIssue https://github.com/dotnet/runtime/issues/53591 for content on GET/HEAD
- more granular ActiveIssue https://github.com/dotnet/runtime/issues/53874 for HttpRequestMessage.Headers.Host
- more granular ActiveIssue https://github.com/dotnet/runtime/issues/53872 for NPE on System.Net.Http.BrowserHttpHandler
- fix HTTP vs HTTPS test configuration `Http2SecureRemoteEchoServer`
- include echo middleware in xharness server
- include middleware in Helix correlation payload

pavelsavara committed:
document https://github.com/dotnet/runtime/issues/53876

pavelsavara commented:
> As @mdh1418 asked, `What does this mean? Is m.Headers.Host problematic?`, could you just explain the reasoning? What fails, and how does it fail?

Created active issue instead https://github.com/dotnet/runtime/issues/53874

pavelsavara committed:
whitespace

radical commented on src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs:
```diff
@@ -343,18 +343,23 @@ public async Task SendAsync_RequestHeaderInResponse_Success(string name, string
                 // External servers do not support HTTP3 currently.
                 return;
             }
+            if (PlatformDetection.IsBrowser && withPort)
+            {
+                // we already have custom port with the local echo server
+                return;
+            }
```
Could you please explain the reasoning for this?

pavelsavara replied:
We are passing `localhost:45678` including the port as server address. Which means we could not append another port to it on local echo server.

radical replied:
`private static readonly string DefaultHttp2AzureServer = "corefx-net-http2.azurewebsites.net";`

`public static string Http2SecureHost => GetValue("DOTNET_TEST_HTTP2SECUREHOST", DefaultHttp2AzureServer);`

So, `DefaultHttp2AzureServer` doesn't have a port number. But the value that we are setting for `DOTNET_TEST_HTTP2SECUREHOST` with XHarness, does? That doesn't sound right. It changes the tests! We should have the same thing for the value from XHarness too. You could split off the port, when reading the value, if present.

And then this change wouldn't be needed either.

pavelsavara replied:
After discussion we agreed on adding `Configuration.Http.SecurePort` and comment about the port number in the env variables. Also relevant fix in this test.

radical commented on src/libraries/System.Net.WebSockets.Client/tests/System.Net.WebSockets.Client.Tests.csproj:
```diff
@@ -5,10 +5,26 @@
     <DefineConstants>$(DefineConstants);NETSTANDARD</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
+    <Scenario>WasmTestOnBrowser</Scenario>
```
same as next line

radical commented on src/libraries/System.Net.WebSockets.Client/tests/System.Net.WebSockets.Client.Tests.csproj:
```diff
@@ -5,10 +5,26 @@
     <Scenario>WasmTestOnBrowser</Scenario>
     <TestArchiveTestsRoot>$(TestArchiveRoot)browseronly/</TestArchiveTestsRoot>
     <TestArchiveTestsDir>$(TestArchiveTestsRoot)$(OSPlatformConfig)/</TestArchiveTestsDir>
+
+    <!-- handle different path to middleware in Helix -->
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' == 'Windows_NT'">%HELIX_CORRELATION_PAYLOAD%/xharness/TestEchoMiddleware</_TestEchoMiddleware>
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' != 'Windows_NT'">$HELIX_CORRELATION_PAYLOAD/xharness/TestEchoMiddleware</_TestEchoMiddleware>
+    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' != 'true'">$(ArtifactsDir)bin/NetCoreServer/$(NetCoreAppCurrent)-$(Configuration)</_TestEchoMiddleware>
+
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --web-server-use-cors --web-server-use-https</WasmXHarnessArgs>
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --set-web-server-http-env=DOTNET_TEST_WEBSOCKETHOST,DOTNET_TEST_HTTPHOST,DOTNET_TEST_HTTP2HOST</WasmXHarnessArgs>
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --set-web-server-https-env=DOTNET_TEST_SECUREWEBSOCKETHOST,DOTNET_TEST_SECUREHTTPHOST,DOTNET_TEST_HTTP2SECUREHOST</WasmXHarnessArgs>
+    <WasmXHarnessArgs>$(WasmXHarnessArgs) --web-server-middleware=$(_TestEchoMiddleware)/NetCoreServer.dll,GenericHandler</WasmXHarnessArgs>
   </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
+    <ProjectReference Include="$(CommonTestPath)System/Net/Prerequisites/NetCoreServer/NetCoreServer.csproj" ReferenceOutputAssembly="false"/>
+  </ItemGroup>
+
```
This can be moved to a separate targets file, and then imported in this, and the other project.

pavelsavara replied:
done

radical commented on src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:
```diff
@@ -59,6 +60,7 @@ public static partial class PlatformDetection
         public static bool IsBrowserDomSupported => GetIsBrowserDomSupported();
         public static bool IsBrowserDomSupportedOrNotBrowser => IsNotBrowser || GetIsBrowserDomSupported();
         public static bool IsNotBrowserDomSupported => !IsBrowserDomSupported;
+        public const TestPlatforms LocalEchoServerIsNotAvailable = TestPlatforms.Any & ~TestPlatforms.Browser;
```
Maybe this should be named like `RemoteEchoServerAvailable`? Because the local echo server can be available for any project. So, something like: 

```csharp
public static bool RemoteEchoServerAvailable => !IsBrowser;
```

And correspondingly:

```csharp
[OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.RemoteEchoServerAvailable)]
```

.. following the pattern used for `PlatformDetection` in the tests.

pavelsavara replied:
done

pavelsavara committed:
feedback

radical commented on src/libraries/Common/tests/System/Net/Configuration.Http.cs:
```diff
@@ -12,11 +12,14 @@ public static partial class Http
             public static string Http2Host => GetValue("DOTNET_TEST_HTTP2HOST", DefaultHttp2AzureServer);
+            public static string Http2SecureHost => GetValue("DOTNET_TEST_HTTP2SECUREHOST", DefaultHttp2AzureServer);
+
+            public static int Port = GetPortValue("DOTNET_TEST_HTTPHOST", 80);
+            public static int SecurePort = GetPortValue("DOTNET_TEST_SECUREHTTPHOST", 443);
```
`DOTNET_TEST_HTTPHOST` -> `DOTNET_TEST_HTTP2HOST`

It would be better to extract the port number from `Http2Host`, and `Http2SecureHost` strings instead. And name these `Http2Port`, `Http2SecurePort`?

radical replied:
Oh, the echo server is `httpHost`, and `http2Host`. This is fine then, my bad. You could extract the value from `Host`, and `SecureHost`, but that's a nit.

pavelsavara replied:
The usage is together with `SecureHost` like `withPort ? Configuration.Http.SecureHost + ":" + Configuration.Http.SecurePort`

ManickaP commented on src/libraries/System.Net.WebSockets.Client/tests/System.Net.WebSockets.Client.Tests.csproj:
```diff
@@ -5,10 +5,26 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
+    <Scenario>WasmTestOnBrowser</Scenario>
     <Scenario>WasmTestOnBrowser</Scenario>
```
Is this line duplicate on purpose?

ManickaP reviewed (commented):
Looks good, just a question about the split of secure server for H/2. I'll approve once it's cleared.

pavelsavara committed:
revert unnecessary rename

pavelsavara committed:
more rename revert

pavelsavara commented on src/libraries/Common/tests/System/Net/Configuration.Http.cs:
```diff
@@ -17,6 +17,7 @@ public static partial class Http
             public static string SecureHost => GetValue("DOTNET_TEST_SECUREHTTPHOST", DefaultAzureServer);
 
             public static string Http2Host => GetValue("DOTNET_TEST_HTTP2HOST", DefaultHttp2AzureServer);
+            public static string Http2SecureHost => GetValue("DOTNET_TEST_HTTP2SECUREHOST", DefaultHttp2AzureServer);
```
I reverted the whole rename and set the DOTNET_TEST_HTTP2HOST env variable as to HTTPS port. Makes simpler PR with less impact to existing assumptions. Thanks everyone for insisting! 

On the other hand, the bad naming of `Http2Host` could confuse the next guy too.

ManickaP replied:
And it will, it's a rite of passage now :smile:

ManickaP reviewed (approved):
Thanks!

radical reviewed (approved):
Thanks for your patience, LGTM! 👍 
You might want to run the outerloop tests to make sure that this didn't break anything.

pavelsavara commented:
/azp run runtime-libraries-mono outerloop
